tap-agent/src/agent/sender_allocation.rs

```diff
@@ -696,31 +705,53 @@ impl SenderAllocationState {
                     error!("Failed to recover receipt signer: {}", e);
                     anyhow!(e)
                 })?;
-            sqlx::query!(
-                r#"
-                    INSERT INTO scalar_tap_receipts_invalid (
-                        signer_address,
-                        signature,
-                        allocation_id,
-                        timestamp_ns,
-                        nonce,
-                        value,
-                        error_log
-                    )
-                    VALUES ($1, $2, $3, $4, $5, $6, $7)
-                "#,
-                receipt_signer.encode_hex(),
-                encoded_signature,
+            debug!(
+                "Receipt for allocation {} and signer {} failed reason: {}",
                 allocation_id.encode_hex(),
-                BigDecimal::from(receipt.message.timestamp_ns),
-                BigDecimal::from(receipt.message.nonce),
-                BigDecimal::from(BigInt::from(receipt.message.value)),
+                receipt_signer.encode_hex(),
                 receipt_error
-            )
-            .execute(&self.pgpool)
-            .await
-            .map_err(|e| anyhow!("Failed to store invalid receipt: {:?}", e))?;
+            );
+            reciepts_signers.push(receipt_signer.encode_hex());
+            encoded_signatures.push(encoded_signature);
+            allocation_ids.push(allocation_id.encode_hex());
+            timestamps.push(BigDecimal::from(receipt.message.timestamp_ns));
+            nounces.push(BigDecimal::from(receipt.message.nonce));
+            values.push(BigDecimal::from(BigInt::from(receipt.message.value)));
+            error_logs.push(receipt_error);
         }
+        sqlx::query!(
+            r#"INSERT INTO scalar_tap_receipts_invalid (
+                signer_address,
+                signature,
+                allocation_id,
+                timestamp_ns,
+                nonce,
+                value,
+                error_log
+            ) SELECT * FROM UNNEST(
+                $1::CHAR(40)[],
+                $2::BYTEA[],
+                $3::CHAR(40)[],
+                $4::NUMERIC(20)[],
+                $5::NUMERIC(20)[],
+                $6::NUMERIC(40)[],
+                $7::TEXT[]
+            )"#,
+            &reciepts_signers,
+            &encoded_signatures,
+            &allocation_ids,
+            &timestamps,
+            &nounces,
+            &values,
+            &error_logs
+        )
+        .execute(&self.pgpool)
+        .await
+        .map_err(|e| {
+            error!("Failed to store receipt: {}", e);
```

Comment:
Could you update this error to something like `"Failed to store invalid receipt: {}"`

Reply:
sure